### ISSUE TITLE
Handle behaviour of  booking time components

### ISSUE THREAD

**Issue by menor** [resmio/widget#69]
Right now it's a bit messy, need to work on:
- What to do with date and timeslot when covers change
- What to do with covers and timeslot when date changes
- What to do with covers when timeslot changes.

**Comment by menor** [comment]
Talked to @fabiennepimenta , but maybe she and @mmarder can check this out:

## 1 - Default Behaviour  
1. 2 **people** is selected 
2. **Calendar** is expanded with today date selected
3. **Timeslots**  
      A. There are availabilities on that day => We show the timeslot selector with no time preselected.  
      B. The restaurant is closed => we show 'closed today' in the calendar component.  
      C. The restaurant is open but there are no availabilities => we show 'No availabilities' in the timeslot component.  

## 2- People Selector changes  
1. **Calendar** selector does not change.
2. **Timeslot** selector changes:  
    A. There are no availabilities for that timeslot but there are on some other timeslot today => ?  
    B. There are no availabilities for that number of persons in the whole day => ?  

## 3- Calendar changes  
1. **People** selector does not change.
2. **Timeslot** selector:  
   A. There's no timeslot  selected yet => We don't change it  
   B. There's a timeslot selected and we have availabilities for that number of persons/date = > We keep the same timeslot selected for the new date  
  C. There's a timeslot selected but we don't have availabilities in the new date => Same as case 2.2  

On 1.3.C I would show the closed in the timeslot component, if not we need to decide what to show in that component in that case.

Need instructions for case 2.2.  

PS: This felt like writing one of these:
![](https://upload.wikimedia.org/wikipedia/en/f/f0/Cave_of_time.jpg)

**Comment by fabiennepimenta** [comment]
2.2 
  A. There are no availabilities for that timeslot but there are on some other timeslot today => We show the available times (basically time opens) and user can select one time;
  B. There are no availabilities for that number of persons in the whole day => We show 'No availabilities'

**Comment by menor** [comment]
But in 1.3.A we have a time preselected, that means that we can change the time the user selected for another one with him not realizing. I wouldn't preselect a time by default anyways, not even in 1.3.A just maybe focus the timeslot selector in that group but with no time preselected.

**Comment by fabiennepimenta** [comment]
Yes, you are right. About the 1.3.A I'm not sure.. Michael said to try with a time pre selected.

**Comment by menor** [comment]
Here are the defaults provided by @neocortex [here](https://github.com/resmio/widget/issues/74)
Date: Today
Covers: 2
Time: 7pm

**Comment by menor** [comment]
Edited to change 1.2 calendar is now collapsed by default

**Comment by menor** [comment]
@mmarder @fabiennepimenta what do we do in 1.3.A if most popular timeslot has no availabilities? I guess going to 2.2.A (We show the available times (basically time opens) and user can select one time) right?

**Comment by fabiennepimenta** [comment]
We can do that or select the second time most used that's : 18h00 and so one..

**Comment by menor** [comment]
Can we make that an 'enhancement'? ;)

**Comment by fabiennepimenta** [comment]
I think so ;)
